packages/protocol/contracts/identity/Attestations.sol

```diff
@@ -649,6 +654,34 @@ contract Attestations is
     }
   }
 
+  /**
+   * @notice Transfer an attestation identifier mapping from the sender address to a replacement address.
+   * @param identifier The hash of the identifier for this attestation.
+   * @param index The index of the account in the accounts array.
+   * @param replacementAddress The address to replace the sender address in the indifier mapping.
+   * @dev Throws if index is out of bound for account array.
+   & @dev Throws if msg.sender is not in the account array at the given index.
+   */
+  function transferAttestationMapping(bytes32 identifier, uint256 index, address replacementAddress)
+    external
+  {
+    uint256 numAccounts = identifiers[identifier].accounts.length;
+    require(index < numAccounts, "Index is invalid");
+    require(
+      msg.sender == identifiers[identifier].accounts[index],
+      "Index does not match msg.sender"
+    );
+    identifiers[identifier].accounts[index] = replacementAddress;
+    AttestedAddress memory attestedAddress = identifiers[identifier].attestations[msg.sender];
```

Comment:
nit: Do we need to keep these in memory? Or could we do something like:
```
identifiers[identifier].attestations[replacementAddress] = identifiers[identifier].attestations[msg.sender];
delete identifiers[identifier].attestations[msg.sender];
```

Reply:
ya that seems doable